README.md

```diff
@@ -1,106 +1,193 @@
 
 ### Introduction
 
-[The "executive summary" or "abstract". Explain in a few sentences what the goals of the project are, and a brief overview of how the solution works. This should be no more than 1-2 paragraphs.]
+WASI Key-Value Store is a WASI API primarily for accessing a key-value datastore. It has functions to:
+
+1. retrieve the value stored and associated with a key,
+2. delete a key-value pair, and
+3. create and update a key-value pair.
+
+The API is designed to hide data-plane differences between different key-value stores. The control-plane behavior/functionality (e.g., including cluster management, data consistency, replication, and sharding) are not specified, and are provider specific.
+
+### Goals
 
-### Goals [or Motivating Use Cases, or Scenarios]
+The primary goal of this API is to allow users to use WASI programs to access key-value stores. It abstracts away the complexity of building stateful services that need to access key-value stores. In other words, it decouples the application code from the specific key-value store APIs. This allows WASI programs to be portable across different key-value store that supports this interface, be it on-prem, in the cloud, or in edge devices.
 
-[What is the end-user need which this project aims to address?]
+The second goal of this API is to abstract away the network stack, allowing applications that need to access key-value stores to not worry about what network protocol is used to access the key-value store.
 
 ### Non-goals
 
-[If there are "adjacent" goals which may appear to be in scope but aren't, enumerate them here. This section may be fleshed out as your design progresses and you encounter necessary technical and other trade-offs.]
+- Cover all application requirements for a key-value store. Instead, it focuses on the most common use cases. It allows providers to extend the API to support more use cases.
+- Transactional semantics.
+- Data consistency.
+- Data replication.
+- Data sharding.
+- Cluster management.
+- Monitoring
 
 ### API walk-through
 
-[Walk through of how someone would use this API.]
+#### Use case 1
 
-#### [Use case 1]
+Imagine you have an HTTP handler that needs to persist some data to a key-value store. The handler needs to be able to retrieve, delete, and update the data in the key-value store. The following Rust code shows how you can use the WASI Key-Value Store API to in the handler.
 
-[Provide example code snippets and diagrams explaining how the API would be used to solve the given problem]
+```rust
+let value = b"value1";
+kv1::create_or_update("key", value)?;
+let val = kv1::get("key")?;
+assert_eq!(val, value);
 
-#### [Use case 2]
-
-[etc.]
+kv1::delete("key")?;
+```
 
 ### Detailed design discussion
 
-[This section should mostly refer to the .wit.md file that specifies the API. This section is for any discussion of the choices made in the API which don't make sense to document in the spec file itself.]
+```go
+interface "wasi:kv/data/crud" {
```

Comment:
For broader audience awareness I think we should avoid the acronym `crud`. While many of us know what create-retrieve-update-delete (crud) means, it might not be all that culturally transferrable.

Since this interface exposes the "basic" read/write functionality, perhaps change it to `wasi:kv/data/basic` or `wasi:kv/data/readwrite` etc.